android/src/main/java/com/applovin/reactnative/AppLovinMAXModule.java

```diff
@@ -648,6 +709,34 @@ public void setTargetingDataKeywords(@Nullable final ReadableArray rawKeywords)
         sdk.getTargetingData().setKeywords( getStringArrayList( rawKeywords ) );
     }
 
+    @ReactMethod
+    public void getTargetingDataKeywords(final Promise promise)
+    {
+        if ( sdk == null )
+        {
+            if ( targetingKeywordsToSet != null && targetingKeywordsToSet.size() > 0 )
+            {
+                promise.resolve( Arguments.fromList( Arguments.toList( targetingKeywordsToSet ) ) );
+            }
+            else
+            {
+                promise.resolve( null );
+            }
+            return;
+        }
+
+        List<String> keywords = sdk.getTargetingData().getKeywords();
+
+        if ( keywords != null && keywords.size() > 0 )
+        {
+            promise.resolve( Arguments.makeNativeArray( keywords.toArray( new String[0] ) ) );
+        }
+        else
+        {
+            promise.resolve( null );
+        }
+    }
+
```

Comment:
This method is overly complicated (e.g. `fromList()` and `toList()` in same line). This is my suggestion, can you help test exactly as I provide it?

```
    @ReactMethod
    public void getTargetingDataKeywords(final Promise promise)
    {
        if ( sdk == null )
        {
            promise.resolve( targetingKeywordsToSet );
            return;
        }

        List<String> keywords = sdk.getTargetingData().getKeywords();

        if ( keywords != null && keywords.size() > 0 )
        {
            promise.resolve( Arguments.makeNativeArray( keywords ) );
        }
        else
        {
            promise.resolve( null );
        }
    }
```

Reply:
This gets compiled, but the first one generates a runtime error.

```
ReactNative     com...lovin.enterprise.apps.demoapp  E  CatalystInstanceImpl caught native exception
       java.lang.RuntimeException: Cannot convert argument of type class com.facebook.react.bridge.ReadableNativeArray
         at com.facebook.react.bridge.Arguments.fromJavaArgs(Arguments.java:183)
         at com.facebook.react.bridge.CallbackImpl.invoke(CallbackImpl.java:31)
         at com.facebook.react.bridge.PromiseImpl.resolve(PromiseImpl.java:56)
         at com.applovin.reactnative.AppLovinMAXModule.getTargetingDataKeywords(AppLovinMAXModule.java:738)
         at java.lang.reflect.Method.invoke(Native Method)
         at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
         at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:151)
         at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
         at android.os.Handler.handleCallback(Handler.java:938)
         at android.os.Handler.dispatchMessage(Handler.java:99)
         at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
         at android.os.Looper.loopOnce(Looper.java:201)
         at android.os.Looper.loop(Looper.java:288)
         at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
         at java.lang.Thread.run(Thread.java:920)
ReactNative     com...lovin.enterprise.apps.demoapp  E  Exception in native call
       java.lang.RuntimeException: Cannot convert argument of type class com.facebook.react.bridge.ReadableNativeArray
         at com.facebook.react.bridge.Arguments.fromJavaArgs(Arguments.java:183)
         at com.facebook.react.bridge.CallbackImpl.invoke(CallbackImpl.java:31)
         at com.facebook.react.bridge.PromiseImpl.resolve(PromiseImpl.java:56)
         at com.applovin.reactnative.AppLovinMAXModule.getTargetingDataKeywords(AppLovinMAXModule.java:738)
         at java.lang.reflect.Method.invoke(Native Method)
         at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
         at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:151)
         at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
         at android.os.Handler.handleCallback(Handler.java:938)
         at android.os.Handler.dispatchMessage(Handler.java:99)
         at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
         at android.os.Looper.loopOnce(Looper.java:201)
         at android.os.Looper.loop(Looper.java:288)
         at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
         at java.lang.Thread.run(Thread.java:920)
```

Reply:
Hmm let me see if I can think of something